core/hyphenator-liang.lua

```diff
@@ -95,11 +95,27 @@ SILE.hyphenator = {}
 SILE.hyphenator.languages = {}
 SILE._hyphenators = {}
 
+-- BUG ALERT: This implementation is known to not account for font.hyphenchar being changed midway through a document.
```

Comment:
But doesn't it occur, if one switches fonts? The dimension the hyphenchar might be different then.

Reply:
If one switches fonts then the node.options is going to be different anyway. The only way this would trigger is if the setting is changed while *not* changing the font while still in the same chunk of text.

But nevermind that. There is another bug lurking and I've backed this out for now.